vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -135,6 +153,22 @@ public long getTotalPages() {
         return (imageRepo.count() % pageSize==0) ? imageRepo.count() / pageSize:(imageRepo.count() / pageSize) + 1;
     }
 
+    /**
+     * Method to return the total image count
+     * 
+     * @return total count of images in DB
+     */
+    @Override
+    public long getTotalImageCountForSelectedFilter(ImageCategory category) {
+        return imageRepo.findByCategoriesLike(category).size();
+    }
+
+    @Override
+    public long getTotalPagesForSelectedFilter(ImageCategory category) {
+        int count = imageRepo.findByCategoriesLike(category).size();
+        return (count % pageSize==0) ? count / pageSize:(count / pageSize) + 1;
```

Comment:
let's change the spacing so it's clear what belongs together
`coutn/pageSize : (count/pageSize)+1;`